gdk/common/exceptions/InitError.py

```diff
@@ -0,0 +1,37 @@
+class InitException(Exception):
+    def __init__(self, reason="", solution="", exception=None, help_text=""):
+        err_details = ""
+        if exception:
+            err_details = f"Error details: {exception}\n"
+        self.message = f"{reason}\n{err_details}{solution}{help_text}"
+        super().__init__(self.message)
+
+
+class InvalidInitArgumentsException(InitException):
+    def __init__(self, exception=None):
+        reason = "The arguments passed with the command are invalid."
+        solution = "Please initialize the project with correct arguments."
+        help_text = "\n\nTry `gdk component init --help`"
+        super().__init__(reason, solution, exception, help_text)
```

Comment:
This can inherit the generic base exception instead. 

The solution and help text could come as arguments as those seem to be specific here. I would suggest to make an overloaded constructor with generic defaults as well.